src/pages/SearchPage/SearchPage.js

```
@@ -6,21 +6,22 @@ import Spinner from '../../components/Spinner/Spinner';
 
 function SearchPage() {
   const history = useHistory();
-
   const [isLoading, setIsLoading] = useState(false);
 
-  async function fetchData() {
+  async function fetchData(searchTerm) {
+    const timeNow = new Date();
+    const timeLastYear = Math.round(timeNow.getTime() / 1000) - 31536000; // Seconds in year
     setIsLoading(true);
     const response = await fetch(
-      'https://www.reddit.com/r/javascript/search.json?q=oop&limit=5',
+      `https://api.pushshift.io/reddit/search/submission/?subreddit=${searchTerm}&sort=desc&sort_type=created_utc&after=${timeLastYear}&size=500`,
       { mode: 'cors' },
     );
     if (response.status === 200) {
```

Checking the status shouldn't be necessary. Afaik fetch should throw an error if something went wrong during the request. Simply `return resonse.json()` in any case.

That said you have to add a try catch block around in handleSearch

```

try {
  const data = await fetchData(searchTerm);
} catch (error) {
  ...
} finally {
  setIsLoading(false)
}
```